arangod/IResearch/IResearchDocument.cpp

```diff
@@ -475,20 +380,28 @@ Field& Field::operator=(Field&& rhs) {
 // --SECTION--                                     FieldIterator implementation
 // ----------------------------------------------------------------------------
 
-/*static*/ FieldIterator const FieldIterator::END;
-
-FieldIterator::FieldIterator()
-  : _name(BufferPool.emplace().release()) { // FIXME don't use shared_ptr
+FieldIterator::FieldIterator(arangodb::transaction::Methods& trx)
+  : _nameBuffer(BufferPool.emplace().release()), // FIXME don't use shared_ptr
```

Comment:
The pattern `.emplace().release()` looks kind of strange to me.
Does it mean a shared ptr is created and emplaced into the pool and then release, so there will be a nullptr shared_ptr in the pool afterwards?

Reply:
`BufferPool.emplace()` returns control object `BufferPool::ptr`. It can transfer ownership to `std::shared_ptr` using `release` method

Reply:
ok